lib/src/main/java/io/ably/lib/util/Crypto.java

```diff
@@ -201,6 +201,7 @@ private CBCCipher(CipherParams params) throws AblyException {
 				encryptCipher = Cipher.getInstance(transformation);
 				encryptCipher.init(Cipher.ENCRYPT_MODE, params.keySpec, params.ivSpec);
 				decryptCipher = Cipher.getInstance(transformation);
+				decryptCipher.init(Cipher.DECRYPT_MODE, params.keySpec, params.ivSpec);
```

Comment:
Why initialise both modes here when you're only going to use one of them?

Reply:
@paddybyers done https://github.com/ably/ably-java/pull/388/commits/864268a24b5482fd26406dc68bc7f778456bcbe7